### PR TITLE
[fleet] add fleet auto upgrade profile commands

### PR DESCRIPTION
**Pull request by pdaru**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az fleet autoupgradeprofile ...`

adding new fleet autoupgradeprofile api commands

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
fleet

File: src/fleet/azext_fleet/_help.py [hunk 1]
```diff
@@ -340,3 +340,48 @@
...
+"""
+
+helps['fleet autoupgradeprofile create'] = """
+    type: command
+    short-summary: Creates or updates an auto upgrade profile.
+    examples:
+        - name: Create an auto upgrade profile with channel 'Rapid' and disabled by default.
+          text: az fleet autoupgradeprofile create -g MyFleetResourceGroup -l MyLocation -f MyFleetName -n MyAutoUpgradeProfileName -c Rapid --disabled
+        - name: Create an auto upgrade profile with channel 'NodeImage'.
+          text: az fleet autoupgradeprofile create -g MyFleetResourceGroup -l MyLocation -f MyFleetName -n MyAutoUpgradeProfileName -c NodeImage --node-image-selection-type Latest
```
Comment (serbrech) [inline src/fleet/azext_fleet/_help.py:356]: fix the example to match the service validation on node-image-selection-type (cannot be set for nodeimage channel)

File: src/fleet/azext_fleet/_help.py [hunk 2]
```diff
@@ -340,3 +340,48 @@
...
+    type: group
+    short-summary: Commands to manage auto upgrade profiles.
+"""
+
+helps['fleet autoupgradeprofile create'] = """
+    type: command
+    short-summary: Creates or updates an auto upgrade profile.
+    examples:
+        - name: Create an auto upgrade profile with channel 'Rapid' and disabled by default.
+          text: az fleet autoupgradeprofile create -g MyFleetResourceGroup -l MyLocation -f MyFleetName -n MyAutoUpgradeProfileName -c Rapid --disabled
```
Comment (serbrech) [inline src/fleet/azext_fleet/_help.py:354]: sub resource do not have a location arguments.

File: src/fleet/azext_fleet/_help.py [hunk 3]
```diff
@@ -340,3 +340,48 @@
...
+        - name: Create an auto upgrade profile with channel 'NodeImage'.
+          text: az fleet autoupgradeprofile create -g MyFleetResourceGroup -l MyLocation -f MyFleetName -n MyAutoUpgradeProfileName -c NodeImage
```
Comment (serbrech) [inline src/fleet/azext_fleet/_help.py:356]: no location argument

File: src/fleet/azext_fleet/_help.py [hunk 4]
```diff
@@ -340,3 +340,48 @@
...
+"""
+
+helps['fleet autoupgradeprofile delete'] = """
+    type: command
+    short-summary: Deletes an auto upgrade profile.
+    examples:
+        - name: Delete an auto upgrade profile.
+          text: az fleet autoupgradeprofile delete -g MyFleetResourceGroup -l MyLocation -f MyFleetName -n MyAutoUpgradeProfileName
```
Comment (serbrech) [inline src/fleet/azext_fleet/_help.py:364]: no location argument

**Comment by pdaru** [comment]
@serbrech & @kairu-ms - PR's is ready for review.

File: src/fleet/HISTORY.rst [hunk 5]
```diff
@@ -107,3 +107,7 @@ Release History
 ++++++
 * Added missing help text
 * Removed dependency on msrestazure library
+
+1.2.3
```
Comment (serbrech) [inline src/fleet/HISTORY.rst:111]: I think this should be 1.3.0?

**Comment by serbrech** [comment]
> ### Module: fleet
> * Update `VERSION` to `1.3.0` in `src/fleet/setup.py`

@pdaru

**Review by kairu-ms — APPROVED** [review]

**Review by serbrech — APPROVED** [review]
LGTM 🚀

**Pull request merged**